File: Cpp/Problem_1544.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

// Problem - 1544: Make The String Great
// C++ Solution!
class Solution {
public:
    string makeGood(string s) {
        stack<char> st;
        string res = "";
        for (char i : s) {
            if (!st.empty() && abs(st.top() - i) == ('a' - 'A'))
                st.pop();
            else
                st.push(i);
        }

        while (!st.empty()) {
            res = st.top() + res;
            st.pop();
        }

        return res;
    }
};

int main() {
    Solution sol;

    cout << sol.makeGood("leEeetcode") << endl;
    cout << sol.makeGood("abBAcC") << endl;
    cout << sol.makeGood("s") << endl;

    return 0;
}

File: Python/Problem_1544.py
# Problem - 1544: Make The String Great
# Python3 Solution!
class Solution:
    def makeGood(self, s: str) -> str:
        res = []
        for i in s:
            if res and abs(ord(res[-1]) - ord(i)) == 32:
                res.pop()
            else:
                res.append(i)

        return "".join(res)

if __name__=="__main__":
    sol = Solution()

    print(sol.makeGood("leEeetcode"))
    print(sol.makeGood("abBAcC"))
    print(sol.makeGood("s"))